src/cartesian/CartesianGrid.js

```diff
@@ -131,6 +136,81 @@ class CartesianGrid extends Component {
     return <g className="recharts-cartesian-grid-vertical">{items}</g>;
   }
 
+  /**
+   * Draw vertical grid stripes filled by colors
+   * @param {Array} verticalPoints either passed in as props or generated from function
+   * @return {Group} Vertical stripes
+   */
+  renderVerticalStripes(verticalPoints) {
+    const { verticalFill } = this.props;
+    if (!verticalFill || !verticalFill.length) { return null; }
+
+    const { fillOpacityVertical, x, y, width, height } = this.props;
+    let verticalPointsUpdated = verticalPoints;
+    if (x === 0) {
```

Comment:
I can't understand this line. Why not unshift x when x !== verticalPoints[0]?

Reply:
Ok, I need to check your suggestion how it works. Thanks!